packages/mermaid/src/diagrams/git/gitGraphAst.ts

```diff
@@ -0,0 +1,510 @@
+import { log } from '../../logger.js';
+import { random } from '../../utils.js';
+import common from '../common/common.js';
+import {
+  setAccTitle,
+  getAccTitle,
+  getAccDescription,
+  setAccDescription,
+  clear as commonClear,
+  setDiagramTitle,
+  getDiagramTitle,
+} from '../common/commonDb.js';
+import type { DiagramOrientation, Commit, GitGraphDB, CommitType } from './gitGraphTypes.js';
+import { ImperativeState } from '../../utils/imperativeState.js';
+
+import DEFAULT_CONFIG from '../../defaultConfig.js';
+import type { GitGraphDiagramConfig } from '../../config.type.js';
+interface GitGraphState {
+  commits: Map<string, Commit>;
+  head: Commit | null;
+  branchConfig: Map<string, { name: string; order: number | undefined }>;
+  branches: Map<string, string | null>;
+  currBranch: string;
+  direction: DiagramOrientation;
+  seq: number;
+  options: any;
+}
+
+const DEFAULT_GITGRAPH_CONFIG: Required<GitGraphDiagramConfig> = DEFAULT_CONFIG.gitGraph;
+
+const mainBranchName = DEFAULT_GITGRAPH_CONFIG.mainBranchName;
+const mainBranchOrder = DEFAULT_GITGRAPH_CONFIG.mainBranchOrder;
+const config: Required<GitGraphDiagramConfig> = structuredClone(DEFAULT_GITGRAPH_CONFIG);
+
+const getConfig = (): Required<GitGraphDiagramConfig> => structuredClone(config);
```

Comment:
If the branch name is changed in frontmatter or init function, it will not be reflected right?
We should merge both the default config, along with the user defined config. 

You can refer https://github.com/mermaid-js/mermaid/blob/develop/packages/mermaid/src/diagrams/packet/db.ts

Also, use assignWithDepth, instead of structuredClone, as it's not supported in some browsers.

Reply:
I understand how to merge using the mergeAndClean function. How do I go about this using assignWithDepth?

Reply:
I think I fixed it, but double check